packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -624,7 +629,8 @@ Hardhat Network's forking functionality only works with blocks from at least spu
 
   public async runCall(
     call: CallParams,
-    blockNumberOrPending: bigint | "pending"
+    blockNumberOrPending: bigint | "pending",
+    stateOverrideSet: OptionalStateOverrideSet = {}
```

Comment:
@fvictorio  does it make sense to have here the optional parameter 'stateOverrideSet'?
Because this parameter is only used with the RPC method eth_call and I have the feeling that I'm 'polluting" the function definition with an additional parameter just for one specific scenario